src/components/textInput/index.tsx

```diff
@@ -0,0 +1,29 @@
+import { TextField, TextFieldProps } from '@mui/material'
+import { Control, Controller } from 'react-hook-form'
+
+type Props = TextFieldProps & {
+  name: 'email' | 'password'
+  control: Control<{ email: string; password: string }, any>
+  label: string
+}
+
+const TextInput = function (props: Props) {
+  const { name, control, label, ...textFieldProps } = props
+
+  return (
+    <Controller
+      name={name}
+      control={control}
+      render={({ field: { onChange, value } }) => (
+        <TextField
+          onChange={onChange}
+          value={value}
+          label={label}
+          {...textFieldProps}
+        />
+      )}
+    />
+  )
+}
+
+export default TextInput
```

Comment:
1. `name` nie może przyjmować tylko dwóch wartości "email" lub "password" - chcemy, żeby był to `string`, bo wtedy możemy przekazać inne nazwy danego pola np. `confirmPassword` w przyszłości

2. Nie potrzebujemy tutaj `control` ponieważ `Form` jest używany z `FormContextProvider` który przekazuje do wszystkich inputów wewnątrz niego odpowiedni `control`

3. Nigdzie nie chcemy `any` w TypeScriptcie 

4. Skorzystaj z `fieldState.error` który ma w sobie pole `message`. W połączeniu z [TextField](https://mui.com/material-ui/react-text-field/#validation) możemy wykorzystać niektóre props'y żeby pokolorować input na czerwono oraz żeby dać wiadomość poniżej

```suggestion
import { TextField, TextFieldProps } from '@mui/material'
import { Controller } from 'react-hook-form'

type Props = TextFieldProps & {
  name: string
  label: string
}

const TextInput = function (props: Props) {
  const { name, label, ...textFieldProps } = props

  return (
    <Controller
      name={name}
      render={({ field: { onChange, value }, fieldState: {error} }) => (
        <TextField
          onChange={onChange}
          value={value}
          label={label}
          ... // tutaj coś trzeba dodać z wykorzystaniem `error`
          {...textFieldProps}
        />
      )}
    />
  )
}

export default TextInput
```